src/Feldspar/Onnx/Operators.hs

```diff
@@ -259,8 +261,8 @@ onnxBatchNormalization attrs xs gamma beta mean var = ys
         epsilon = value $ P.realToFrac $ getAttr attrs aaFloat 1e-5 "epsilon"
 
 -- | Flatten a tensor to a matrix
-onnxFlatten :: Pushy vec => Attrs -> vec a -> Push DIM2 a
-onnxFlatten attrs vec = flatPush (P.fromIntegral $ getAttr attrs aaInt 1 "axis") $ toPush vec
+onnxFlatten :: (Pushy vec, Syntax a) => Attrs -> vec a -> Pull DIM2 a
+onnxFlatten attrs vec = toPull $ store $ flatPush (P.fromIntegral $ getAttr attrs aaInt 1 "axis") $ toPush vec
```

Comment:
The comment from the previous pull request about efficiency would be useful to have here.

Reply:
Ok!